waspc/cli/src/Wasp/Cli/Question.hs

```diff
@@ -0,0 +1,86 @@
+module Wasp.Cli.Question
+  ( query,
+    queryWithOptions,
+    queryNotNull,
+  )
+where
+
+import Data.List (intercalate)
+import qualified Data.Text as T
+import System.IO (hFlush, stdout)
+import Text.Read (readMaybe)
+import qualified Wasp.Util.Terminal as Term
+
+queryNotNull :: String -> IO String
+queryNotNull question = do
+  ensureNotNull . query $ question
+
+queryWithOptions :: String -> [String] -> IO (Maybe String)
+queryWithOptions question options = do
+  putStrLn optionsStr
+  putStr $ question ++ " "
+  answer <- prompt
+
+  handleEmptyAnswer answer
+    >>= \case
+      Just _ ->
+        handleOptionIndex answer
+          >>= \case
+            Just option -> return $ Just option
+            Nothing -> handleOptionName answer
+      -- Optional
+      Nothing -> return Nothing
```

Comment:
So I haven't tried to understand it fully, but I think you can make this somewhat simpler by possibly making more functions pure + using the short circuiting capability of the Maybe Monad. I am not 100% sure, but it looks too convoluted at the moment.

If you want to give it a try on your own with this suggestion in mind, give it a try, and if you run out of ideas and it still looks complex, we can try it together.

Reply:
I'm not sure how to use
> using the short circuiting capability of the Maybe Monad
in this context, I'll push the code I have and if you can help me with it 🤷